code/workspaces/auth-service/src/dataaccess/userRolesRepository.js

```diff
@@ -45,4 +46,19 @@ async function addRole(userId, projectName, role) {
   return roleAdded;
 }
 
-export default { getRoles, getProjectUsers, addRole };
+async function removeRole(userId, projectName) {
+  const query = { userId };
+  const user = await UserRoles().findOne(query).exec();
+
+  if (user) {
+    const { projectRoles } = user;
+    const roleIndex = findIndex(projectRoles, { projectName });
+    if (roleIndex > -1) {
+      remove(projectRoles, { projectName });
+      await UserRoles()
+        .findOneAndUpdate(query, user, { upsert: true, setDefaultsOnInsert: true, runValidators: true });
```

Comment:
Will this work for removal too?

Reply:
Do you mean the `findOneAndUpdate`? Yes, it will work as we're not removing the whole record just the role from the user.